### PR TITLE
feat: consul

### PR DESCRIPTION
**Pull request by antmat**
Hub failover over consul.
Still need some fixes:
  * sometimes leader is not being discovered by miner
  * what to do with ip determination?
  * ...
  * PROFIT!

@3Hren @noxiouz PTAL

### CODE REVIEW

File: insonmnia/hub/server.go [hunk 1]
```diff
@@ -227,6 +275,17 @@ func (h *Hub) applyFilters(miner *MinerCtx, req *pb.TaskRequirements) (bool, err
 
 // StartTask schedules the Task on some miner
 func (h *Hub) StartTask(ctx context.Context, request *pb.HubStartTaskRequest) (*pb.HubStartTaskReply, error) {
+	if !h.isLeader {
```
Comment (3Hren) [inline insonmnia/hub/server.go]: Seems like boilerplate. Maybe rewrite using reflection to simple forward the request without looking into it?

**Comment by 3Hren** [comment]
News?

**Comment by Sokel** [comment]
ready to merge?

**Comment by antmat** [comment]
@Sokel wanna TAL?

**Comment by Sokel** [comment]
@antmat no, thats all

**Pull request merged**